backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/InlineClasses.kt

```diff
@@ -96,7 +101,13 @@ enum class KonanPrimitiveType(val classId: ClassId, val binaryType: BinaryType.P
     val fqName: FqNameUnsafe get() = this.classId.asSingleFqName().toUnsafe()
 
     companion object {
-        val byFqName = KonanPrimitiveType.values().associateBy { it.fqName }
+        val byFqNameParts = KonanPrimitiveType.values().groupingBy { it.classId.packageFqName }
+                .aggregate { _, accumulator: MutableMap<Name, KonanPrimitiveType>?, element, first ->
+                    if (first)
+                        mutableMapOf(element.classId.shortClassName to element)
+                    else
+                        accumulator!!.also { it[element.classId.shortClassName] = element }
```

Comment:
The logic selecting key-value pair is duplicated. Can using `fold` help to make it simplier?
If not, then `groupBy {}.mapValues { associate }` sequence will be simplier than `groupingBy` approach.